docs: upgrade testnet documentation (#1949)
FuelLabs/fuels-ts: apps/docs/src/guide/introduction/getting-started.md

At a high level, you can use the Fuel TypeScript SDK to build applications that can run computations on the Fuel Virtual Machine through interactions with smart contracts written in Sway.

For this interaction to work, the SDK must be able to communicate with a [`fuel-core`](https://github.com/FuelLabs/fuel-core) node; you have two options at your disposal:

1. Connecting to the [Testnet](#connecting-to-the-testnet). (For application building)
2. Running a [local node](https://docs.fuel.network/guides/running-a-node/). (For smart contract testing)

### Connecting to the Testnet

> The **Testnet** is a public network that allows you to interact with a Fuel Virtual Machine (managed by Fuel) and is used predominantly for application building.

Reviewer: ```suggestion
The **Testnet** is a public network that allows you to interact with a Fuel Virtual Machine and is used for testing and development purposes.
```

> [!NOTE] Latest Testnet
> Beta 5
>
> `https://beta-5.fuel.network/graphql`

We have some useful resources for the Testnet:

- [**Faucet**](https://faucet-beta-5.fuel.network/) - for funding wallets that have been created.

> - [**Explorer**](https://next-app.fuel.network/) - for viewing transactions and blocks.

Reviewer 2: Is the `next-app` url correct? Should it be `app`?

Author: 0d5db132896f4046e4141147de2d578d9c0b35bb

> - [**Explorer**](https://next-app.fuel.network/) - for viewing transactions and blocks.

Author: ```suggestion
- [**Explorer**](https://app.fuel.network/) - for viewing transactions and blocks.
```

> - [**Playground**](https://beta-5.fuel.network/playground) - for testing GraphQL queries and mutations.

Reviewer 2: ```suggestion
- [**GraqphQL Playground**](https://beta-5.fuel.network/playground) - for testing GraphQL queries and mutations.
```

We have sway playground as well. Probably best to differentiate.

---

> With the example below, we connect a [Provider](../providers/index.md) to the Latest Testnet node and create a new wallet from a private key.

Reviewer 3: ```suggestion
In the example below, we connect a [Provider](../providers/index.md) to the latest testnet and create a new wallet from a private key.
```

- "In the example below" is IMO better than "With the example below". "With" feels too prescriptive. 
- "Latest Testnet node" -> "latest testnet node"
    - Latest Testnet shouldn't be capitalized
    - The testnet is a collection of nodes so it's best to just not mention "node" at all

> **Note:** New wallets on the Testnet will not have any assets! You can use the [Faucet](https://faucet-beta-5.fuel.network/) to fund your wallet.

<<< @/../../docs-snippets/src/guide/introduction/getting-started.test.ts#connecting-to-the-testnet{ts:line-numbers}

### Connecting to a local node

If you want to connect to another node just change the URL or IP and port. For example, to connect to a local node that was created with `fuel-core` you can use:

<<< @/../../../packages/fuel-gauge/src/doc-examples.test.ts#provider-local{ts:line-numbers}

## Further Resources and Next Steps

For a more in-depth, step-by-step guide on working with the wider Fuels ecosystem, check out the [Developer Quickstart](https://docs.fuel.network/guides/quickstart/). This guide covers:

> 1. Installing all tools needed to develop with the Fuels ecosystem.

Author: 4146f4802fead5055f58addd00de2780c002a197
